File: frontend/src/services/donationService.js
import api from './api';

const donationService = {
  getAllDonations: async (page = 0, size = 10) => {
    const response = await api.get(`/payments?page=${page}&size=${size}`);
    return response.data;
  },

  getDonationStats: async () => {
    const response = await api.get('/payments/stats');
    return response.data;
  },

  getDonationById: async (id) => {
    const response = await api.get(`/payments/${id}`);
    return response.data;
  },

  createDonation: async (data) => {
    const response = await api.post('/payments', data);
    return response.data;
  },

  getDonationsByType: async (type, page = 0, size = 10) => {
    const response = await api.get(`/payments/type/${type}?page=${page}&size=${size}`);
    return response.data;
  }
};

export default donationService;

File: backend/src/main/java/com/masjid/app/service/DonationService.java
package com.masjid.app.service;

import com.masjid.app.entity.Donation;
import com.masjid.app.repository.DonationRepository;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.PageRequest;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;
import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.HashMap;
import java.util.Map;

@Service
public class DonationService {

    private final DonationRepository donationRepository;

    public DonationService(DonationRepository donationRepository) {
        this.donationRepository = donationRepository;
    }

    public Page<Donation> getAllDonations(int page, int size) {
        Pageable pageable = PageRequest.of(page, size);
        return donationRepository.findAllByOrderByCreatedAtDesc(pageable);
    }

    public Donation createDonation(Donation donation) {
        return donationRepository.save(donation);
    }

    public Map<String, Object> getDonationStats() {
        LocalDateTime weekAgo = LocalDateTime.now().minusDays(7);
        LocalDateTime monthAgo = LocalDateTime.now().minusDays(30);

        Map<String, Object> stats = new HashMap<>();
        stats.put("weeklyTotal", donationRepository.sumTotalSince(weekAgo));
        stats.put("monthlyTotal", donationRepository.sumTotalSince(monthAgo));
        stats.put("weeklyZakat", donationRepository.sumZakatSince(weekAgo));
        stats.put("weeklyDonations", donationRepository.sumDonationsSince(weekAgo));
        stats.put("totalBalance", getTotalBalance());

        return stats;
    }

    public BigDecimal getTotalBalance() {
        LocalDateTime yearAgo = LocalDateTime.now().minusYears(1);
        return donationRepository.sumTotalSince(yearAgo);
    }
}